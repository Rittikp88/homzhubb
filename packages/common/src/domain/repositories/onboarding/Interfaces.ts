export interface IOnboardingRepository {
  getDetails(): Promise<any>;
}
