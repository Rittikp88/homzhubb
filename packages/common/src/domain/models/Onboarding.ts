import { JsonObject, JsonProperty } from '@homzhub/common/src/utils/ObjectMapper';

@JsonObject('Onboarding')
export class Onboarding {
  @JsonProperty('title', String)
  private _title = '';

  @JsonProperty('description', String)
  private _description = '';

  @JsonProperty('image_url', String)
  private _image_url = '';

  get title(): string {
    return this._title;
  }

  get description(): string {
    return this._description;
  }

  get image_url(): string {
    return this._image_url;
  }
}
