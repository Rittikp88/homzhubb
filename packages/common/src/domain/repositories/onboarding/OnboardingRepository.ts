import { BootstrapAppService } from '@homzhub/common/src/services/BootstrapAppService';
import { IApiClient } from '@homzhub/common/src/network/Interfaces';
import { IOnboardingRepository } from '@homzhub/common/src/domain/repositories/onboarding/Interfaces';

const ENDPOINTS = {
  allDetail: (): string => 'https://jsonplaceholder.typicode.com/users',
};

class OnboardingRepository implements IOnboardingRepository {
  private apiClient: IApiClient;

  public constructor() {
    this.apiClient = BootstrapAppService.clientInstance;
  }

  public getDetails = async (): Promise<any> => {
    const url = ENDPOINTS.allDetail(); // TODO: Replace OnboardingData with url once api is ready
    return await this.apiClient.get(url);
  };
}

const onboardingRepository = new OnboardingRepository();
export { onboardingRepository as OnboardingRepository };
