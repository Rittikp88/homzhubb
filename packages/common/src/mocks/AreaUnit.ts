export const AreaUnit = [
  {
    label: 'Sq.ft',
    value: 'Sq.ft',
  },
  {
    label: 'Sq.yard',
    value: 'Sq.yard',
  },
  {
    label: 'Sq.m',
    value: 'Sq.m',
  },
  {
    label: 'cents',
    value: 'cents',
  },
  {
    label: 'biswa',
    value: 'biswa',
  },
  {
    label: 'acres',
    value: 'acres',
  },
  {
    label: 'hectares',
    value: 'hectares',
  },
  {
    label: 'kattha',
    value: 'kattha',
  },
];
