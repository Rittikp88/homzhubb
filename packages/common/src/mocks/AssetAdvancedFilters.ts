export const AdvancedFilters = {
  searchRadius: [
    {
      value: 1,
      label: 'This area only',
    },
    {
      value: 2,
      label: 'Within 0.25 miles',
    },
    {
      value: 3,
      label: 'Within 0.5 miles',
    },
    {
      value: 4,
      label: 'Within 1 miles',
    },
    {
      value: 5,
      label: 'Within 3 miles',
    },
    {
      value: 6,
      label: 'Within 10 miles',
    },
    {
      value: 7,
      label: 'Within 20 miles',
    },
    {
      value: 8,
      label: 'Within 30 miles',
    },
    {
      value: 9,
      label: 'Within 40 miles',
    },
  ],
  dateAdded: [
    {
      value: 1,
      label: 'Anytime',
    },
    {
      value: 2,
      label: 'Last 3 days',
    },
    {
      value: 3,
      label: 'Last 14 days',
    },
    {
      value: 4,
      label: 'Last 24 hours',
    },
    {
      value: 5,
      label: 'Last 7 days',
    },
    {
      value: 6,
      label: 'Last 28 days',
    },
  ],
  propertyAge: [
    {
      value: 1,
      label: 'Less than 1 year',
    },
    {
      value: 2,
      label: 'Less than 5 years',
    },
    {
      value: 3,
      label: 'Less than 10 years',
    },
    {
      value: 4,
      label: 'Less than 15 years',
    },
    {
      value: 5,
      label: 'Less than 20 years',
    },
    {
      value: 6,
      label: 'More than 20 years',
    },
  ],
  moveInDate: '',
  facing: [
    {
      value: 1,
      title: 'East',
    },
    {
      value: 2,
      title: 'West',
    },
    {
      value: 3,
      title: 'North',
    },
    {
      value: 4,
      title: 'South',
    },
    {
      value: 5,
      title: 'North-East',
    },
    {
      value: 6,
      title: 'South-east',
    },
    {
      value: 7,
      title: 'North-West',
    },
    {
      value: 8,
      title: 'South-West',
    },
    {
      value: 9,
      title: 'Any value',
    },
  ],
  furnishing: [
    {
      value: 1,
      label: 'Television',
    },
    {
      value: 2,
      label: 'Iron',
    },
    {
      value: 3,
      label: 'Washing Machine',
    },
    {
      value: 4,
      label: 'Geysers',
    },
    {
      value: 5,
      label: 'Microwave',
    },
    {
      value: 6,
      label: 'Modular Kitchen',
    },
    {
      value: 7,
      label: 'Beds',
    },
    {
      value: 8,
      label: 'Air-Purifier',
    },
    {
      value: 9,
      label: 'Mattress',
    },
    {
      value: 10,
      label: 'Air Conditioner',
    },
  ],
  propertyAmenities: [
    {
      value: 1,
      label: 'Swimming Pool',
    },
    {
      value: 2,
      label: 'Near Metro station',
    },
    {
      value: 3,
      label: 'Gym',
    },
    {
      value: 4,
      label: 'Tennis court',
    },
    {
      value: 5,
      label: 'Daily needs store',
    },
    {
      value: 6,
      label: 'Table tennis',
    },
    {
      value: 7,
      label: 'Near ATM',
    },
    {
      value: 8,
      label: 'Club-house',
    },
    {
      value: 9,
      label: 'Near petrol pump',
    },
    {
      value: 10,
      label: 'Helipad',
    },
  ],
  isAgentListed: false,
  isVerified: false,
};
