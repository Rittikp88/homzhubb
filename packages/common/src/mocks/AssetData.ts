import { theme } from '@homzhub/common/src/styles/theme';
import { images } from '@homzhub/common/src/assets/images';

export interface IAssetData {
  id: number;
  property_name: string;
  address: string;
  type: string;
  color: string;
  images: any;
  contacts?: any;
  isPropertyCompleted: boolean;
}

export const TenanciesAssetData = [
  {
    id: 5,
    project_name: 'House 2',
    unit_number: '103',
    block_number: 'B',
    latitude: 25.6207722,
    longitude: 85.1280892,
    carpet_area: 2500.0,
    carpet_area_unit: 'SQ_FT',
    floor_number: 2,
    total_floors: 3,
    asset_type: {
      id: 5,
      name: 'Studio Apartment',
    },
    spaces: [
      {
        id: 1,
        name: 'Bedroom',
        count: 2,
      },
      {
        id: 2,
        name: 'Bathroom',
        count: 1,
      },
    ],
    asset_group: {
      id: 1,
      name: 'Residential',
    },
    digital_id: '8fad8389-6893-411a-bd4a-4b1575523d4a',
    construction_year: null,
    is_gated: null,
    description: '',
    furnishing: 'SEMI',
    attachments: [
      {
        file_name: 'prof.jpg',
        is_cover_image: true,
        link: 'https://homzhub-bucket.s3.amazonaws.com/asset_images/8e8c48fc-c089-11ea-8247-34e12d38d70eprof.jpg',
        media_type: 'IMAGE',
        media_attributes: {},
      },
    ],
    verification_documents: [],
    verifications: {
      verification_status: 2,
      description:
        '**What is this verification?**\nSed ut perspiciatis unde omnis iste natus error sit voluptatem\naccusantium doloremque laudantium.\n\nNemo enim ipsam voluptatem quia voluptas sit aspernatur\naut odit aut fugit. Neque porro quisquam est,\nqui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit,\nsed quia non numquam eius modi tempora incidunt ut labore et dolore\nmagnam aliquam quaerat voluptatem.',
    },
    progress_percentage: 74.0,
    notifications: {
      count: 5,
    },
    serviceTickets: {
      count: 0,
    },
    assetStatusInfo: {
      tag: {},
      leaseTransaction: {},
      leaseTenantInfo: {},
    },
  },
];

export const PortfolioAssetData = [
  {
    id: 1,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'OWNER',
    isPropertyCompleted: true,
    color: '#CE9B6C',
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 2,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'RENTED',
    isPropertyCompleted: true,
    color: theme.colors.gradientE,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 3,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'VACANT',
    isPropertyCompleted: true,
    color: theme.colors.highPriority,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 4,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'VACANT',
    isPropertyCompleted: false,
    color: theme.colors.highPriority,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
  },
  {
    id: 5,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'FOR SALE',
    isPropertyCompleted: true,
    color: theme.colors.mediumPriority,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 6,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'MAINTAIN',
    isPropertyCompleted: true,
    color: theme.colors.informational,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
];

export const AssetFilter = [
  {
    label: 'All',
    value: 'all',
  },
  {
    label: 'Rented',
    value: 'rented',
  },
  {
    label: 'Vacant',
    value: 'vacant',
  },
  {
    label: 'For Sale',
    value: 'sale',
  },
  {
    label: 'Maintain',
    value: 'maintain',
  },
];

export const DocumentsData = [
  {
    id: 1,
    name: 'Quarterly Inspection Report',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 2,
    name: 'Rent',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 3,
    name: 'Electricity bill',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 4,
    name: 'Water charges',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 5,
    name: 'Maintenance Charges',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
];
