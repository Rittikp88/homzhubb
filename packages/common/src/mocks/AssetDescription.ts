export const mockReviews = [
  {
    id: 1,
    experience_area: 'Location and Neighbourhood',
    rating: 93,
  },
  {
    id: 2,
    experience_area: 'Fair Pricing for Rent',
    rating: 91,
  },
  {
    id: 3,
    experience_area: 'Property Up-keep and maintenance',
    rating: 49,
  },
];

export const mockAsset = {
  id: 1,
  project_name: '2972 Westheimer Rd. Santa Ana, NY',
  unit_number: '12',
  block_number: '32',
  carpet_area: '123',
  carpet_area_unit: 'Sq.ft',
  latitude: '13.77',
  longitude: '77.833',
  floor_number: 4,
  total_floors: 10,
  posted_on: '21 Aug 2020',
  available_from: 'Immediately',
  description:
    'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32.\n' +
    '\n' +
    'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from "de Finibus Bonorum et Malorum" by Cicero are also reproduced in their exact original form, accompanied by English versions from the 1914 translation by H. Rackham.',
  asset_group: {
    id: 1,
    name: 'Residential',
  },
  asset_type: {
    id: 1,
    name: 'Farm House',
  },
  spaces: [
    {
      id: 1,
      name: 'Bedroom',
      count: 5,
    },
    {
      id: 2,
      name: 'Bathroom',
      count: 10,
    },
  ],
  attachments: [
    {
      id: 1,
      name: 'image1.png',
      link: 'http://image.png',
      attachment_type: 'IMAGE',
      mime_type: 'Image/Jpeg',
      is_cover_image: true,
    },
    {
      id: 2,
      name: 'video1.mp4',
      link: 'http://video.mp4',
      attachment_type: 'VIDEO',
      mime_type: 'video/mp4',
      is_cover_image: false,
    },
  ],
  lease_term: {
    id: 1,
    expected_price: 18888,
    maintenance_paid_by: 'OWNER',
    utility_paid_by: 'OWNER',
    maintenance_amount: 5000,
    maintenance_schedule: 'MONTHLY',
  },
  sale_term: null,
  verification: {
    code: 3,
    description: '',
  },
  features: [
    {
      name: 'Booking Amount',
      locale_key: 'booking_amount',
      value: 10000,
    },
    {
      name: 'Booking Amount',
      locale_key: 'booking_amount',
      value: 10000,
    },
    {
      name: 'Booking Amount',
      locale_key: 'booking_amount',
      value: 10000,
    },
  ],
  amenities: [
    {
      id: 37,
      name: 'Gymnasium',
      category: {
        id: 13,
        name: 'General',
      },
      attachment: {
        id: 39,
        name: 'gym.svg',
        link: 'https://homzhub-bucket.s3.amazonaws.com/Group+2428gym.svg',
      },
    },
    {
      id: 38,
      name: 'Swimming Pool',
      category: {
        id: 13,
        name: 'General',
      },
      attachment: {
        id: 40,
        name: 'pool.svg',
        link: 'https://homzhub-bucket.s3.amazonaws.com/Group+2428gym.svg',
      },
    },
    {
      id: 39,
      name: 'CCTV Security',
      category: {
        id: 13,
        name: 'General',
      },
      attachment: {
        id: 41,
        name: 'cctv.svg',
        link: 'https://homzhub-bucket.s3.amazonaws.com/Group+2428gym.svg',
      },
    },
    {
      id: 40,
      name: 'Parking',
      category: {
        id: 13,
        name: 'General',
      },
      attachment: {
        id: 42,
        name: 'parking.svg',
        link: 'https://homzhub-bucket.s3.amazonaws.com/Group+2428gym.svg',
      },
    },
    {
      id: 41,
      name: 'Wifi',
      category: {
        id: 13,
        name: 'General',
      },
      attachment: {
        id: 43,
        name: 'wifi.svg',
        link: 'https://homzhub-bucket.s3.amazonaws.com/Group+2428gym.svg',
      },
    },
    {
      id: 42,
      name: 'Intercom',
      category: {
        id: 13,
        name: 'General',
      },
      attachment: {
        id: 44,
        name: 'intercom.svg',
        link: 'https://homzhub-bucket.s3.amazonaws.com/Group+2428gym.svg',
      },
    },
    {
      id: 43,
      name: 'Balcony',
      category: {
        id: 13,
        name: 'General',
      },
      attachment: {
        id: 45,
        name: 'balcony.svg',
        link: 'https://homzhub-bucket.s3.amazonaws.com/Group+2428gym.svg',
      },
    },
  ],
  highlights: [
    {
      name: 'Power packup',
      covered: true,
    },
    {
      name: 'Power packup',
      covered: true,
    },
    {
      name: 'Power packup',
      covered: true,
    },
    {
      name: 'Power packup',
      covered: true,
    },
    {
      name: 'Power packup',
      covered: true,
    },
    {
      name: 'Power packup',
      covered: true,
    },
  ],
  contacts: {
    id: 1,
    full_name: 'Anuvrat Somnath',
    email: 'anuvrat.somnath@nineleaps.com',
    country_code: '+91',
    phone_number: '7903470293',
  },
};
