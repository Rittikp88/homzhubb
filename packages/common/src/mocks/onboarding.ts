export const OnboardingData = [
  {
    heading: 'Search',
    description:
      'Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a. Sed massa facilisi vel neque id dolor',
    buttonText: 'Next',
    scene: require('@homzhub/common/src/assets/images/scene.jpeg'),
  },
  {
    heading: 'Post',
    description:
      'Ornare nunc, sagittis est ut. Mi arcu morbi faucibus consectetur auctor. Diam odio sed vel nam ultricies.',
    buttonText: 'Next',
    scene: require('@homzhub/common/src/assets/images/scene1.jpeg'),
  },
  {
    heading: 'Maintain',
    description:
      'Mattis ut est senectus commodo id sem arcu erat luctus. Id egestas eu enim, varius. Ut tellus id neque enim.',
    buttonText: 'Got it',
    scene: require('@homzhub/common/src/assets/images/scene2.jpeg'),
  },
];
