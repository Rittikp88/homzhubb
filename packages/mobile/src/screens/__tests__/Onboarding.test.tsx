import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Onboarding, { mapDispatchToProps } from '@homzhub/mobile/src/screens/OnBoarding';
import { OnboardingData } from '@homzhub/common/src/mocks/onboarding';
import toJson from 'enzyme-to-json';
import { mount, shallow } from 'enzyme';

const createTestProps = (testProps: object): any => ({
  ...testProps,
});
const mock = jest.fn();
const mockStore = configureStore([]);

jest.mock('@homzhub/mobile/src/components/atoms/Carousel', () => 'SnapCarousel');
jest.mock('@homzhub/common/src/components/atoms/Text', () => 'Text');
jest.mock('@homzhub/common/src/components/atoms/Text', () => 'Label');
jest.mock('@homzhub/common/src/components/atoms/Button', () => 'Button');

describe('Onboarding Screen', () => {
  let store;
  let component;
  let props;
  const OnboardingDataExample = {
    data: OnboardingData,
  };

  beforeEach(() => {
    store = mockStore({
      onboarding: OnboardingDataExample,
    });
    props = createTestProps({
      onboarding: OnboardingDataExample,
      activeSlide: 0,
      ref: {},
      getOnboardingDetail: mock,
      navigation: {},
    });
    component = shallow(
      <Provider store={store}>
        <Onboarding {...props} />
      </Provider>
    );
  });

  it('should match snapshot', () => {
    expect(toJson(component.dive())).toMatchSnapshot();
  });

  it('should match snapshot 1', () => {
    const mountedComponent = mount(
      <Provider store={store}>
        <Onboarding {...props} />
      </Provider>
    );
    expect(toJson(mountedComponent)).toMatchSnapshot();
  });

  it('should dispatch actions', () => {
    mapDispatchToProps(mock).getOnboardingDetail();
  });
});
