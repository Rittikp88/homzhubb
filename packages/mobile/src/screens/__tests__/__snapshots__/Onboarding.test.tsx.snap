// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Screen should match snapshot 1`] = `
<Connect(Component)
  activeSlide={0}
  getOnboardingDetail={[MockFunction]}
  navigation={Object {}}
  onboarding={
    Object {
      "data": Array [
        Object {
          "buttonText": "Next",
          "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a. Sed massa facilisi vel neque id dolor",
          "heading": "Search",
          "scene": Object {
            "testUri": "../../../packages/common/src/assets/images/scene.jpeg",
          },
        },
        Object {
          "buttonText": "Next",
          "description": "Ornare nunc, sagittis est ut. Mi arcu morbi faucibus consectetur auctor. Diam odio sed vel nam ultricies.",
          "heading": "Post",
          "scene": Object {
            "testUri": "../../../packages/common/src/assets/images/scene1.jpeg",
          },
        },
        Object {
          "buttonText": "Got it",
          "description": "Mattis ut est senectus commodo id sem arcu erat luctus. Id egestas eu enim, varius. Ut tellus id neque enim.",
          "heading": "Maintain",
          "scene": Object {
            "testUri": "../../../packages/common/src/assets/images/scene2.jpeg",
          },
        },
      ],
    }
  }
/>
`;
